public/app/features/explore/LogsVolumePanelList.tsx

```diff
@@ -67,7 +69,8 @@ export const LogsVolumePanelList = ({
       <SupplementaryResultError
         title="The logs volume query is taking too long"
         suggestion="Continue executing"
-        onSuggestionClicked={retry}
+        onSuggestion={retry}
```

Comment:
Could we replace `retry` and instead add running of actual metric query. Because users can set up custom logs volume timeouts and these can be reason why logs volume query failed (error `context closed`). So instead of hitting timeout again, we'd give them opportunity to run actual metric query that wouldn't be affected by this timeout. 

Because I don't think this will reliably help. Might possibly help if there are already cached results in Loki, but if we'll give users a way to run actual query, we'll definitely help.

Reply:
For sure. Would that be possible from the `Logs volume` panel?

My initial plan was to run a modified version of `retry` but with no timeout, by modifying the current implementation to support that.

Reply:
@ivanahuckova Please take a look at my last commit: https://github.com/grafana/grafana/pull/65100/commits/a142929bbe73b7b7f7aeb80539e1c227d2794fee

Since the logs volume data depends on the supplementary queries definitions and is hooked to the Explore redux state, I came up with that workaround, where I define a new type of supplementary query `LogsVolumeNoTimeout`, which executes a modified version of the `LogsVolume` query but with `supportingQueryType = undefined`, to bypass the implemented timeout.

When this data is received, I also modified `loadSupplementaryQueryData` with an optional parameter `displayAsType`, to display the results of `LogsVolumeNoTimeout` in `LogsVolume`.

I tried several alternatives to arrive to this, let me know what you think, and feel free to suggest other alternatives. Thanks!

Reply:
Thanks for the feedback. 

> What if instead of new supplementary query type that is relevant only for Loki data source, we've added a possibility to specify some optional custom options in getSupplementaryQuery

I would love to do that! The problem is that the queries are already instantiated in the state https://github.com/grafana/grafana/blob/main/public/app/features/explore/state/query.ts#L612-L620 and to get the results we just subscribe https://github.com/grafana/grafana/blob/main/public/app/features/explore/state/query.ts#L772 to the output value of the query, without any possibility of passing an argument, for example, from a call to `loadSupplementaryQueryData()`.

> I am kind of worried that we are giving users a way how to override something that their admins set up on proxy side (admins need to manually create timeout for users running these queries).

That was the request from @slim-bean . If you really want to run this query, even though it's a costly request, you can do it.

I might have an idea on how to implement this (e.g. if `loadSupplementaryQueryData` has extra parameters, we get a new `dataProvider` instance), which will probably require more back and forth between us to decide on the approach.

> query chunking will probably keep queries under timeout limit.

Query chunking is not going GA any time soon, so we shouldn't expect it to improve anything until it's either no longer experimental or the default behavior of the datasource.

What I'm going to do is to use the feedback here to reiterate on this problem, and begin with just creating a dedicated alert for timeouts. Once we agree on that, we can work on another iteration to run the query without timeout.

Thanks!

Reply:
Another solution could be giving users a way to run the same metric query as actual query and not as supplementary/logs volume query.